langmeta/langmeta/shared/src/main/scala/org/langmeta/semanticdb/internal/package.scala

```diff
@@ -62,6 +62,11 @@ package object semanticdb {
     def toDb(sourcepath: Option[Sourcepath], sdoc: s.TextDocument): d.Document = {
       val s.TextDocument(sschema, suri, stext, slanguage, ssymbols, soccurrences, sdiagnostics, ssynthetics) = sdoc
       assert(sschema == s.Schema.SEMANTICDB3, "s.TextDocument.schema must be ${s.Schema.SEMANTICDB3}")
+      def dsymbol(ssymbol: String) = {
+        def suffix = suri.replace("/", "_").replace(".", "_")
```

Comment:
suri is always a file?

Reply:
What if you have a cross-project?

Reply:
`suri` is a relative path (original source's path minus the original sourceroot).

In case of multiple projects sharing the same files, you end up with SemanticDB payloads that have the same uris. This means that local symbols aren't guaranteed to be unique within universes that mix such projects.

However, this problem was also present in SemanticDB v2. This pull requests doesn't aim to fix this problem though - just to achieve feature parity between v2 and v3.